Add torch version constraint to be equal or less than 2.8.0 to prevent breaking onnx changes in torch 2.9.0

samt1995 opened the pull request:
### Proposed change(s)

The new ` 2.9.0` torch version released 2 day ago results in an issue finding `onnxscript`. `ml-agents` is configured to take any version >= `2.1.1` so it will take this version. Full credits to @Nitesh-joestar for identifying the issue and the fix for this in https://github.com/Unity-Technologies/ml-agents/issues/6250.

Adding a constraint that maximum version is `2.8.0` prevents this error.

I'm not exactly of the exact root cause but it seems there were some backwards incompatible changes to onnx: https://github.com/pytorch/pytorch/releases/tag/v2.9.0#backwards-incompatible-changes.

Stack trace on exiting training:
```
  File "C:\Users\REDACTED\miniconda3\envs\mlagents\lib\site-packages\mlagents\trainers\torch_entities\model_serialization.py", line 164, in export_policy_model
    torch.onnx.export(
  File "C:\Users\REDACTED\miniconda3\envs\mlagents\lib\site-packages\torch\onnx\__init__.py", line 282, in export
    from torch.onnx._internal.exporter import _compat
  File "C:\Users\REDACTED\miniconda3\envs\mlagents\lib\site-packages\torch\onnx\_internal\exporter\_compat.py", line 16, in <module>
    from torch.onnx._internal.exporter import (
  File "C:\Users\REDACTED\miniconda3\envs\mlagents\lib\site-packages\torch\onnx\_internal\exporter\_core.py", line 18, in <module>
    import onnxscript
ModuleNotFoundError: No module named 'onnxscript'
```
### Useful links (Github issues, JIRA tickets, ML-Agents forum threads etc.)
https://github.com/Unity-Technologies/ml-agents/issues/6250

### Types of change(s)

- [x] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the changelog (if applicable)
- [ ] Updated the documentation (if applicable)
- [ ] Updated the migration guide (if applicable)

### Other comments
Tested with torch version 2.8.0, running training, then exiting training; and confirm file saved correctly:
```
[INFO] Exported results\5\Basic\Basic-269.onnx
[INFO] Copied results\5\Basic\Basic-269.onnx to results\5\Basic.onnx.
```

samt1995 committed:
Pin torch version to 2.8.0 to prevent breaking onnx changes in 2.9.0

The ` 2.9.0` version results in an issue finding `onnxscript`. Pinning the `2.8.0` prevent this error. 

I'm not exactly of the exact root cause but it seems there were some backwards incompatible changes to onnx: https://github.com/pytorch/pytorch/releases/tag/v2.9.0#backwards-incompatible-changes.

Stack trace on exiting training:
```
  File "C:\Users\Samuel\miniconda3\envs\mlagents\lib\site-packages\mlagents\trainers\torch_entities\model_serialization.py", line 164, in export_policy_model
    torch.onnx.export(
  File "C:\Users\Samuel\miniconda3\envs\mlagents\lib\site-packages\torch\onnx\__init__.py", line 282, in export
    from torch.onnx._internal.exporter import _compat
  File "C:\Users\Samuel\miniconda3\envs\mlagents\lib\site-packages\torch\onnx\_internal\exporter\_compat.py", line 16, in <module>
    from torch.onnx._internal.exporter import (
  File "C:\Users\Samuel\miniconda3\envs\mlagents\lib\site-packages\torch\onnx\_internal\exporter\_core.py", line 18, in <module>
    import onnxscript
ModuleNotFoundError: No module named 'onnxscript'
```

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/Unity-Technologies/ml-agents?pullRequest=6251) <br/>All committers have signed the CLA.

samt1995 committed:
Add torch version constraint to be equal or less than 2.8.0

Previously commit forced only 2.8.0, however adding this constraint is a bit nicer, as I see in some of installation docs sometimes torch 2.2.0 is installed: https://github.com/Unity-Technologies/ml-agents/blob/release_22_docs/docs/Installation.md.

maryamziaa reviewed (approved):
Hi, Thanks for submitting the PR. Please add a change log:
https://github.com/Unity-Technologies/ml-agents/commit/c11149b7a465116951f43cb25ad3761374c7fe18

Other than that, it looks good to me!

samt1995 committed:
Update CHANGELOG.md

samt1995 commented:
No worries, have added the change log.
